contracts/Safe.sol

```diff
@@ -312,7 +310,10 @@ contract Safe is
                     // The signature data for contract signatures is appended to the concatenated signatures and the offset is stored in s
                     contractSignature := add(add(signatures, s), 0x20)
                 }
-                require(ISignatureValidator(currentOwner).isValidSignature(data, contractSignature) == EIP1271_MAGIC_VALUE, "GS024");
+                require(
+                    ISignatureValidator(currentOwner).isValidSignature(abi.encode(dataHash), contractSignature) == EIP1271_MAGIC_VALUE,
```

Comment:
Since we're already breaking things, why did we decide not to change the `ISignatureValidator` interface and remove the legacy method?

Reply:
Also, this will always pass `bytes` to the validator, won't it? I thought our goal was to get rid of this

Reply:
I was thinking on same to remove old method as well.